src/binxml/value_variant.rs

```diff
@@ -263,13 +263,11 @@ impl<'a> BinXmlValue<'a> {
             (BinXmlValueType::Real64Type, _) => BinXmlValue::Real64Type(try_read!(cursor, f64)?),
             (BinXmlValueType::BoolType, _) => BinXmlValue::BoolType(try_read!(cursor, bool)?),
             (BinXmlValueType::GuidType, _) => BinXmlValue::GuidType(try_read!(cursor, guid)?),
-            // TODO: find a sample with this token.
-            (BinXmlValueType::SizeTType, _) => {
-                return Err(DeserializationError::UnimplementedValueVariant {
-                    name: "SizeT".to_owned(),
-                    size,
-                    offset: cursor.position(),
-                })
+            (BinXmlValueType::SizeTType, Some(4)) => {
+                BinXmlValue::HexInt32Type(try_read!(cursor, hex32)?)
+            }
+            (BinXmlValueType::SizeTType, Some(8)) => {
+                BinXmlValue::HexInt64Type(try_read!(cursor, hex64)?)
             }
```

Comment:
Happy to do whatever you would like for this as its your library. The reason I did not keep the old error is that I thought the catch all at the bottom would be sufficient now, as it would be there for edge cases? But I guess if we do that the error type is now redundant and needs to be removed?

Reply:
It will catch it, but keeping the special match arm gives a better error so it's a bit nicer (as we already have an error just for that)